recipes/metall/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+import os
+
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import rmdir
+
+required_conan_version = ">=1.33.0"
+
+
+class Recipe(ConanFile):
+    name = "metall"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/LLNL/metall"
+    description = "Meta allocator for persistent memory"
+    license = "MIT", "Apache-2.0"
+    topics = "cpp", "allocator", "memory-allocator", "persistent-memory", "ecp", "exascale-computing"
+    settings = "build_type", "compiler", "os", "arch"
+    options = {
+        # Disable freeing file space
+        "disable_free_file_space": [True, False],
+        # Disable small object cache
+        "disable_small_object_cache": [True, False],
+        # Use VM space aware algorithm in the bin directory
+        "use_sorted_bin": [True, False],
+        # Set the default VM reserve size (use the internally defined value if 0 is specified)
+        "default_vm_reserve_size": "ANY",
+        # Set the max segment size (use the internally defined value if 0 is specified)
+        "max_segment_size": "ANY",
+        # Set the initial segment size (use the internally defined value if 0 is specified)
+        "initial_segment_size": "ANY",
+        # Experimental: Use the anonymous map when creating a new map region
+        "use_anonymous_new_map": [True, False],
+        # Experimental: Build programs that require the NUMA policy library (numa.h)
+        "build_numa": [True, False],
+        # Experimental: Try to free the associated pages and file space when objects equal to or larger than that
+        # is deallocated
+        "free_small_object_size_hint": "ANY",
+    }
+
+    default_options = {
+        "boost:header_only": True,
+        "disable_free_file_space": False,
+        "disable_small_object_cache": False,
+        "use_sorted_bin": False,
+        "default_vm_reserve_size": 0,
+        "max_segment_size": 0,
+        "initial_segment_size": 0,
+        "use_anonymous_new_map": False,
+        "build_numa": False,
+        "free_small_object_size_hint": 0,
+    }
+    requires = "boost/1.75.0",
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self.folders.base_source,
+                  strip_root=True)
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8.3",
+            "clang": "9",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._min_cppstd)
+        if self.settings.os == "Windows":
```

Comment:
```suggestion
        if self.settings.os != "Linux":
```
According the message error, requires Linux

Reply:
I've double checked the code. It has preprocessor switches for Linux and Macos and requires some POSIX functionalities (especially mmap). I would say, allowing only Linux and Macos is reasonable.